spec/AsComponent.coffee

```diff
@@ -0,0 +1,252 @@
+          done()
+      it 'should send to OUT port', (done) ->
+        wrapped = noflo.asCallback 'ascomponent/sync-default',
+          loader: loader
+        wrapped
+          name: 'Maailma'
+        , (err, res) ->
+          return done err if err
+          chai.expect(res).to.eql
+            out: 'Hello Maailma'
+          done()
+  describe 'with a function returning a Promise', ->
+    describe 'with a resolved promise', ->
+      before ->
+        @skip() if isBrowser and typeof window.Promise is 'undefined'
+      func = (hello) ->
+        return new Promise (resolve, reject) ->
+          setTimeout ->
+            resolve "Hello #{hello}"
+          , 5
+      it 'should be possible to componentize', (done) ->
+        component = -> noflo.asComponent func
+        loader.registerComponent 'ascomponent', 'promise-one', component, done
+      it 'should send to OUT port', (done) ->
+        wrapped = noflo.asCallback 'ascomponent/promise-one',
+          loader: loader
+        wrapped 'World', (err, res) ->
+          return done err if err
+          chai.expect(res).to.equal 'Hello World'
+          done()
+    describe 'with a rejected promise', ->
+      before ->
+        if isBrowser and typeof window.Promise is 'undefined'
+          return @skip()
+      func = (hello) ->
+        return new Promise (resolve, reject) ->
+          setTimeout ->
+            reject new Error "Hello #{hello}"
+          , 5
+      it 'should be possible to componentize', (done) ->
+        component = -> noflo.asComponent func
+        loader.registerComponent 'ascomponent', 'sync-throw', component, done
+      it 'should send to ERROR port', (done) ->
+        wrapped = noflo.asCallback 'ascomponent/sync-throw',
+          loader: loader
+        wrapped 'Error', (err) ->
+          chai.expect(err).to.be.an 'error'
+          chai.expect(err.message).to.equal 'Hello Error'
+          done()
+  describe 'with a synchronous function taking zero parameters', ->
+    describe 'with returned value', ->
+      func = () ->
+        return "Hello there"
+      it 'should be possible to componentize', (done) ->
+        component = -> noflo.asComponent func
+        loader.registerComponent 'ascomponent', 'sync-zero', component, done
+      it 'should contain correct ports', (done) ->
+        loader.load 'ascomponent/sync-zero', (err, instance) ->
+          return done err if err
+          chai.expect(Object.keys(instance.inPorts.ports)).to.eql ['in']
+          chai.expect(Object.keys(instance.outPorts.ports)).to.eql ['out', 'error']
+          done()
+      it 'should send to OUT port', (done) ->
+        wrapped = noflo.asCallback 'ascomponent/sync-zero',
+          loader: loader
+        wrapped 'bang', (err, res) ->
+          return done err if err
+          chai.expect(res).to.equal 'Hello there'
+          done()
+    describe 'with a built-in function', ->
```

Comment:
This only works for functions that don't take any arguments? Test description suggests otherwise... And considering this limination, isn't it almost better if it just throws Error explaining that function arguments cant be introspected - and one should create a wrapper function with arguments?

Reply:
@jonnor I guess we should just document it.

Reply:
Ok